### DISCUSSION TITLE
Where is written to check, when is a player putting a shulker box into a shulker box?

### DISCUSSION THREAD

**Discussion by AgentKosticka** [FabricMC/fabric-api#2436] [Mod Dev Support]
I want to create a mod with simmilar behaviour like putting a shulker box into shulker box, but i can't seem to find it. I don't know where to search else than external libraries -> net.minecraft:... -> ... -> block -> shulker box block or -> inventories. If you know it would make my day

**Comment by warjort** [comment]
The key to this is understanding how the loot/ItemStack is created when you break the block.
The shulker box's contents are stored in the ItemStack and of course an ItemStack can be stored in any container.

Look at ShulkerBoxBlock.getDroppedStacks()
but the important part is how the loot table is defined to use dynamic drops
see data/minecraft/loot_tables/blocks/shulker_box.json in the minecraft jar

**Reply by warjort** [reply]
The top part of that that loot table is for additional state associated with your block entity, like the shulker box's lock state or custom name

**Reply by AgentKosticka** [reply]
Thanks, but I am trying to add more items to the check and remove the shulker box from the check. How would i do that?

**Reply by warjort** [reply]
I don't understand what you are asking. sorry.

You need to show what you are doing and include all the relevant information.

Try to avoid using unqualified "it" and "the" and describe what you are actually referring to.

**Reply by AgentKosticka** [reply]
so when you click to place item into a shulker box, minecraft checks what is that item, it's count, nbt etc. and if the item is minecraft:shulker_box it disallows you to place the shulker box inside another shulker box. I want to make a mod, that overwrites this check to allow it's users to place shulker box inside another shulker box. But i don't know how the mechanics of checking if the item is a shulker box work.

**Reply by warjort** [reply]
See haykam821's answer.

**Comment by haykam821** [comment]
The items that can be nested are controlled by the `ShulkerBoxBlockEntity#canInsert` and the `BlockItem#canBeNested` methods. Do note that the `BlockItem#canBeNested` method also affects bundles.
